client/my-sites/checkout/controller.jsx

```diff
@@ -70,6 +71,19 @@ export function checkout( context, next ) {
 		context.pathname.includes( '/checkout/no-site' ) &&
 		'no-user' === context.query.cart;
 
+	const searchParams = new URLSearchParams( window.location.search );
+	const isSignupCheckout = searchParams.get( 'signup' ) === '1';
+
+	if ( isSignupCheckout && ! isDomainOnlyFlow ) {
+		window.addEventListener( 'beforeunload', function () {
+			const signupDestinationCookieExists = retrieveSignupDestination();
+
+			if ( signupDestinationCookieExists ) {
+				sessionStorage.setItem( 'signupCheckoutPageUnloaded', true ); // eslint-disable-line no-undef
```

Comment:
Just like what you've done with the cookie, I'd suggest to abstract the access to this session storage flag as a pair of getter/setter in utils.js, so we can both name them as meaningful functions and keep them in one place for easier maintenance.

Reply:
Done - I've moved sessionStorage getters and setters to their own utils file.